presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -13,202 +13,105 @@
-            leftPositions = new int[leftPositionCount * 2];
+        if (Math.max(rightArray.getPositionCount(), leftArray.getPositionCount()) >= NESTED_LOOP_THRESHOLD) {
+            return arraysOverlapSetBased(elementType, leftArray, rightArray);
         }
 
-        if (rightPositions == null || rightPositions.length < rightPositionCount) {
-            rightPositions = new int[rightPositionCount * 2];
-        }
-
-        for (int i = 0; i < leftPositionCount; i++) {
-            leftPositions[i] = i;
-        }
-        for (int i = 0; i < rightPositionCount; i++) {
-            rightPositions[i] = i;
-        }
-        IntArrays.quickSort(leftPositions, 0, leftPositionCount, intBlockCompare(type, leftArray));
-        IntArrays.quickSort(rightPositions, 0, rightPositionCount, intBlockCompare(type, rightArray));
-
-        int leftCurrentPosition = 0;
-        int rightCurrentPosition = 0;
-        while (leftCurrentPosition < leftPositionCount && rightCurrentPosition < rightPositionCount) {
-            if (leftArray.isNull(leftPositions[leftCurrentPosition]) || rightArray.isNull(rightPositions[rightCurrentPosition])) {
-                // Nulls are in the end of the array. Non-null elements do not overlap.
-                return null;
+        boolean hasNull = false;
+        for (int i = 0; i < leftArray.getPositionCount(); i++) {
+            if (leftArray.isNull(i)) {
+                hasNull = true;
+                continue;
             }
-
-            int compareValue;
             try {
-                compareValue = type.compareTo(leftArray, leftPositions[leftCurrentPosition], rightArray, rightPositions[rightCurrentPosition]);
-            }
-            catch (NotSupportedException e) {
-                throw new PrestoException(NOT_SUPPORTED, e.getMessage(), e);
-            }
-
-            if (compareValue > 0) {
-                rightCurrentPosition++;
-            }
-            else if (compareValue < 0) {
-                leftCurrentPosition++;
+                for (int j = 0; j < rightArray.getPositionCount(); j++) {
+                    if (rightArray.isNull(j)) {
+                        hasNull = true;
+                        continue;
+                    }
+                    if (elementType.equalTo(leftArray, i, rightArray, j)) {
+                        return true;
+                    }
+                }
             }
-            else {
-                return true;
+            catch (Throwable t) {
+                throw internalError(t);
             }
         }
-        return leftArray.isNull(leftPositions[leftPositionCount - 1]) || rightArray.isNull(rightPositions[rightPositionCount - 1]) ? null : false;
+        if (hasNull) {
+            return null;
+        }
+        return false;
     }
 
-    public Boolean genericArraysOverlap(Block leftArray, Block rightArray, AbstractType type)
+    public static Boolean arraysOverlapSetBased(Type type,
```

Comment:
This should be a private function right? If you make it private, you also don't need to check for 0 position count, or checkArgument that position count is at least `NESTED_LOOP_THRESHOLD`.